sql/src/test/java/io/crate/metadata/rule/ingest/IngestRulesMetaDataTest.java

```diff
@@ -80,4 +84,45 @@ public void testXContentSerialization() throws IOException {
         // a metadata custom must consume the surrounded END_OBJECT token, no token must be left
         assertThat(parser.nextToken(), nullValue());
     }
+
+    @Test
+    public void testGetIngestRulesForValidSourceReturnsIngestRules() {
+        Set<IngestRule> ingestRules = inputMetaData.getIngestRules(SOURCE_NAME);
+        assertThat(ingestRules.size(), is(1));
+        assertThat(ingestRules.iterator().next().getName(), is(RULE_NAME));
+    }
+
+    @Test
+    public void testGetIngestRulesForMissingSourceReturnsNull() {
+        assertThat(inputMetaData.getIngestRules("some_missing_source"), nullValue());
+    }
+
+    @Test
+    public void testCreateIngestRule() {
+        inputMetaData.createIngestRule("new_source", new IngestRule("new_rule", "log", "topic like log%"));
+
+        Set<IngestRule> newSourceRules = inputMetaData.getIngestRules("new_source");
+        assertThat(newSourceRules, notNullValue());
+        assertThat(newSourceRules.size(), is(1));
+        assertThat(newSourceRules.iterator().next().getName(), is("new_rule"));
+    }
+
+    @Test
+    public void testCreateExistingRuleThrowsIllegalArgumentException() {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("Ingest rule with name " + RULE_NAME + " already exist");
+        inputMetaData.createIngestRule("new_source", new IngestRule(RULE_NAME, "log", "topic like log%"));
+    }
+
+    @Test
+    public void testDropExistingIngestRule() {
+        long affectedRow = inputMetaData.dropIngestRule(RULE_NAME);
```

Comment:
seems like this test relies on that the createRule test has run before. this won't be the case always and so it will be flaky and fail. pls use setUp/tearDown methods if you need a state.